userspace/libsinsp/threadinfo.cpp

```diff
@@ -1417,6 +1445,12 @@ void sinsp_thread_manager::increment_mainthread_childcount(sinsp_threadinfo* thr
 	}
 }
 
+std::unique_ptr<sinsp_threadinfo> sinsp_thread_manager::new_threadinfo() const
```

Comment:
why should we use a unique pointer here if we immediately release it after this call?

Reply:
or better I understand your point of trying to use safe pointers instead of raw ones, but this leads to a sort of simple pointer/smart-pointer dance around the code... btw we can keep this as it is right now and maybe one day try to clean up all these usages :thinking:

Reply:
We need to start from somewhere doing the right thing, right? 😂